src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -28,28 +29,79 @@ def get_serials(self):
         return osc.system.serial_port_choices()
 
     def get_disks(self):
-        disks = {}
-        disks_data = self.retrieve_disks_data()
+        ctx = pyudev.Context()
 
-        for dev in pyudev.Context().list_devices(subsystem='block', DEVTYPE='disk'):
-            if dev.sys_name.startswith(('sr', 'md', 'dm-', 'loop', 'zd')):
-                continue
-            if RE_NVME_PRIVATE_NAMESPACE.match(dev.sys_name):
+        disks = {}
+        for dev in ctx.list_devices(subsystem='block', DEVTYPE='disk'):
+            if dev.sys_name.startswith(IGNORE) or RE_NVME_PRIV.match(dev.sys_name):
                 continue
-            device_type = os.path.join('/sys/block', dev.sys_name, 'device/type')
-            if os.path.exists(device_type):
-                with open(device_type, 'r') as f:
-                    if f.read().strip() != '0':
-                        continue
-            # nvme drives won't have this
 
             try:
-                disks[dev.sys_name] = self.get_disk_details(dev, self.disk_default.copy(), disks_data)
+                disks[dev.sys_name] = self.get_disk_details(dev, is_nvme=dev.sys_name.startswith('nvme'))
```

Comment:
Can we calculate `is_nvme` inside `get_disk_details`?

Reply:
Why?

Reply:
To simplify function call to just `self.get_disk_details(dev)`. It has no sense to make caller calculate `is_nvme`. Imagine this
```
self.get_disk_details(dev, is_nvme=dev.sys_name.startswith('nvme'),
                      size=self.safe_retrieval(dev.attributes, 'size', None, asint=True))
```
Pretty weird, right?

Reply:
No, not to me 🤷‍♂️